flink-table/flink-table-planner-blink/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -2670,12 +2670,12 @@ private boolean accept_(RexNode e, List<RexNode> newTerms) {
                             e.getKind(),
                             newTerms);
                 case IS_NULL:
-                    if (negate) {
-                        return false;
-                    }
-                    final RexNode arg = ((RexCall) e).operands.get(0);
-                    return accept1(
-                            arg, e.getKind(), rexBuilder.makeNullLiteral(arg.getType()), newTerms);
+                    /**
+                     *   when is_null in OR operate ,if change to search('',null) there was a major mistake,
+                     *   because search has three return values (true,false,null),but OR Operate only
+                     *   can return true or false,so wo can't change is_null or ... to search('',null);
+                     */
+                    return false;
```

Comment:
Could you also add a test to verify the expected simplified result in `FlinkRexUtilTest`?

Reply:
The changes forbidden the `is not null` simplification overall, which is beyond the scope of the case. A better way it to expand the `SEARCH` operator after the `RexSimplify` invocation, see `FlinkRexUtil.expandSearch`.

Reply:
@danny0405 do you mean,in FlinkRexUtil.expandSearch , i strip out the optimization of is_null?